src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1495,6 +1497,161 @@ static bool extendedTypeIsOrDerivedFrom(NamedTypeSymbol hiddenContainer, NamedTy
             }
         }
 
+        /// <summary>
+        /// We remove non-methods that are "hidden" (based on extended type) by more specific members.
+        /// But we collect all the methods, because we'll only know if they are "hidden" after we have the complete set of extension members.
+        ///
+        /// Once we have the complete set of extension members, we'll call <see cref="CompleteMergeExtensionLookupResultsHidingLessSpecific(LookupResult, ConsList{TypeSymbol}, ref CompoundUseSiteInfo{AssemblySymbol})"/>
+        /// It will remove methods that are "hidden" by any remaining non-methods.
+        /// It will also compact the set (deal with any null entries left in the result).
+        /// </summary>
+        private void MergeExtensionLookupResultsHidingLessSpecificNonMethods(LookupResult current, LookupResult additional, ConsList<TypeSymbol> basesBeingResolved, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            if (!current.IsMultiViable || !additional.IsMultiViable)
+            {
+                current.MergePrioritized(additional);
+                return;
+            }
+
+            var currentSymbols = current.Symbols;
+            var additionalSymbols = additional.Symbols;
+
+            var currentCount = currentSymbols.Count;
+            var additionalCount = additionalSymbols.Count;
+            Debug.Assert(currentCount > 0);
+            Debug.Assert(additionalCount > 0);
+
+            for (int i = 0; i < additionalCount; i++)
+            {
+                var additionalSymbol = additionalSymbols[i];
+
+                removeLessSpecificNonMethodCurrentSymbols(additionalSymbol, currentSymbols, out bool isAdditionalSymbolHidden,
+                    currentCount, basesBeingResolved, ref useSiteInfo);
+
+                if (!isAdditionalSymbolHidden)
+                {
+                    currentSymbols.Add(additionalSymbol);
```

Comment:
>currentSymbols.Add(additionalSymbol); [](http://example.com/codeflow?start=20&length=37)

It looks like we will be checking this symbol against other symbols in the same result. At best this looks unnecessary.